From 64d764b4a5635787ef1827b2bc175b8ae3ed3d26 Mon Sep 17 00:00:00 2001
From: Yan Burman <yanb@mellanox.com>
Date: Thu, 7 Feb 2013 02:25:24 +0000
Subject: [PATCH 515/858] net/mlx4_en: Save previous MAC address of the port so
 we can replace it later

In preparation to having more than one unicast MAC per port, we need to keep track
of the previous MAC address in the flow of ndo_set_mac_address,
so that mlx4_en_replace_mac will know what to replace.

Signed-off-by: Yan Burman <yanb@mellanox.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 90bbb74af68b3255bc298731f8b60d5668877306)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx4/en_netdev.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index d363eba..a6faa3e 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -591,7 +591,7 @@ static void mlx4_en_put_qp(struct mlx4_en_priv *priv)
 }
 
 static int mlx4_en_replace_mac(struct mlx4_en_priv *priv, int qpn,
-          unsigned char *new_mac)
+          unsigned char *new_mac, unsigned char *prev_mac)
 {
  struct mlx4_en_dev *mdev = priv->mdev;
  struct mlx4_dev *dev = mdev->dev;
@@ -657,7 +657,7 @@ static void mlx4_en_do_set_mac(struct work_struct *work)
  if (priv->port_up) {
   /* Remove old MAC and insert the new one */
   err = mlx4_en_replace_mac(priv, priv->base_qpn,
-       priv->dev->dev_addr);
+       priv->dev->dev_addr, priv->prev_mac);
   if (err)
    en_err(priv, "Failed changing HW MAC address\n");
   memcpy(priv->prev_mac, priv->dev->dev_addr,
-- 
1.8.3.1